### PR TITLE
Add support for RTL to Wrap

### LINKED ISSUE #11389: Wrap RTL

### PR DESCRIPTION
**Pull request by abarth**
Fixes #11389

### CODE REVIEW

File: packages/flutter/lib/src/rendering/wrap.dart [hunk 1]
```diff
@@ -235,6 +263,116 @@ class RenderWrap extends RenderBox with ContainerRenderObjectMixin<RenderBox, Wr
     markNeedsLayout();
   }
 
+  /// Determines the order to lay children out horizontally and how to interpret
+  /// `start` and `end` in the horizontal direction.
+  ///
+  /// If the [direction] is [Axis.horizontal], this controls which order
+  /// children are painted in (left-to-right or right-to-left), and the meaning
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/wrap.dart]: just painted, or laid out as well?

File: packages/flutter/lib/src/rendering/wrap.dart [hunk 2]
```diff
@@ -235,6 +263,116 @@ class RenderWrap extends RenderBox with ContainerRenderObjectMixin<RenderBox, Wr
...
+  /// of the [alignment] property's [WrapAlignment.start] and
+  /// [WrapAlignment.end] values.
+  ///
+  /// If the [direction] is [Axis.horizontal], and either the
+  /// [alignment] is either [WrapAlignment.start] or [WrapAlignment.end], or
+  /// there's more than one child, then the [textDirection] must not be null.
+  ///
+  /// If the [direction] is [Axis.vertical], this controls the meaning of the
+  /// [runAlignment] property's [WrapAlignment.start] and
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/wrap.dart]: Does it also control the order of layout in the vertical direction (when the Wrap wraps)?

File: packages/flutter/lib/src/rendering/wrap.dart [hunk 3]
```diff
@@ -235,6 +263,116 @@ class RenderWrap extends RenderBox with ContainerRenderObjectMixin<RenderBox, Wr
...
+  /// If the [direction] is [Axis.horizontal], this controls the meaning of the
+  /// [runAlignment] property's [WrapAlignment.start] and
+  /// [WrapAlignment.end] values, as well as the [crossAxisAlignment] property's
+  /// [WrapCrossAlignment.start] and [WrapCrossAlignment.end] values.
+  ///
+  /// If the [direction] is [Axis.horizontal], and either the
+  /// [runAlignment] is either [WrapAlignment.start] or [WrapAlignment.end], the
+  /// [crossAxisAlignment] is either [WrapCrossAlignment.start] or
+  /// [WrapCrossAlignment.end], or there's more than one child, then the
+  /// [verticalDirection] must not be null.
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/wrap.dart]: apply to these docs whatever you apply to the previous docs

File: packages/flutter/lib/src/widgets/basic.dart [hunk 4]
```diff
@@ -3241,6 +3249,56 @@ class Wrap extends MultiChildRenderObjectWidget {
   ///    other in the cross axis.
   final WrapCrossAlignment crossAxisAlignment;
 
+  /// Determines the order to lay children out horizontally and how to interpret
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/basic.dart]: apply earlier fixes to these two docs too

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/rendering/wrap.dart (hunk 2 above)
Comment (abarth) [reply packages/flutter/lib/src/rendering/wrap.dart]: Yes.

File: packages/flutter/lib/src/rendering/wrap.dart (hunk 1 above)
Comment (abarth) [reply packages/flutter/lib/src/rendering/wrap.dart]: Laid out as well.  I'll update the flex docs too, which is where this text comes from.

File: packages/flutter/lib/src/rendering/wrap.dart (hunk 2 above)
Comment (Hixie) [reply packages/flutter/lib/src/rendering/wrap.dart]: We should mention that too then.

**Pull request merged**